mutiny-core/src/nodemanager.rs

```diff
@@ -739,15 +797,65 @@ impl NodeManager {
     /// Lists all the on-chain transactions in the wallet.
     /// These are sorted by confirmation time.
     pub fn list_onchain(&self) -> Result<Vec<TransactionDetails>, MutinyError> {
-        let mut txs = self.wallet.list_transactions(false)?;
+        let mut txs = self.wallet.list_transactions(true)?;
         txs.sort();
+        let address_labels = self.get_address_labels()?;
+        let txs = txs
+            .into_iter()
+            .map(|tx| {
+                let labels = tx
+                    .transaction
+                    .clone()
+                    .unwrap()
+                    .output
+                    .iter()
+                    .find_map(|o| {
+                        if let Ok(addr) = Address::from_script(&o.script_pubkey, self.network) {
+                            address_labels.get(&addr).cloned()
+                        } else {
+                            None
+                        }
+                    })
+                    .unwrap_or_default();
+
+                TransactionDetails {
+                    labels,
+                    ..tx.into()
```

Comment:
This should be extracted so that both `list_onchain` and `get_transaction` use it. Identical code.